auth/credentials/api.go

```diff
@@ -0,0 +1,70 @@
+package credentials
+
+import (
+	"context"
+	"net/http"
+)
+
+// tokenAPIPath for obtaining OAuth Access token from server
+//
+//nolint:gosec //url only
+const tokenAPIPath = "/api/oauth/token"
+
+// serverURL for atlas API
+const serverURL = "https://cloud.mongodb.com" + tokenAPIPath
+
+type ServiceAccountOAuthClientWithTokenSource struct {
+	ClientID     string
+	ClientSecret string
+	// Custom token source. InMemoryTokenSource being default
+	TokenSource TokenSource
+
+	// Custom context
+	Context *context.Context
+	// Custom base url for change
+	BaseURL *string
+}
+
+// NewServiceAccountOAuthClientWithTokenSource initializes an OAuthClient with client credentials and custom TokenSource
+// Use this method to initialze custom token storage (filesystem)
+func NewServiceAccountOAuthClientWithTokenSource(opts ServiceAccountOAuthClientWithTokenSource) *OAuthClient {
+	var tokenURL string
+	if opts.BaseURL != nil {
+		tokenURL = *opts.BaseURL + tokenAPIPath
+	} else {
+		tokenURL = serverURL + tokenAPIPath
```

Comment:
not sure if I am misreading this, but isn't serverURL already = "https://cloud.mongodb.com" + tokenAPIPath ?

Reply:
Code have been updated. Thank you